lib/Pipeline/Global.cpp

```diff
@@ -18,7 +18,34 @@ using namespace std;
 using namespace pipeline;
 using namespace llvm;
 
-Error Global::storeToDisk(StringRef Path) const {
+llvm::Error
+Global::deserializeCache(const Context &Ctx, const llvm::MemoryBuffer &Buffer) {
+  auto MaybeCache = ReadPathCache::deserialize(Ctx,
+                                               Buffer,
+                                               getSerializationPrimitives());
+  if (not MaybeCache)
+    return MaybeCache.takeError();
+
+  PathCache = std::move(*MaybeCache);
+  return llvm::Error::success();
+}
+
+llvm::Error Global::serializeCache(llvm::raw_ostream &OS) const {
+  return getPathCache().serialize(OS);
+}
+
+Error Global::storeToDisk(StringRef Path, bool StoreCache) const {
```

Comment:
I'd make storing the cache mandatory. Let's not have potentially harmful configuration options unless we have a good reason for them.

Reply:
Let's move to have cache serialized on disk on the side of containers.